lib/DiscoveryStreamFeed.jsm

```diff
@@ -491,6 +529,7 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
           const isBelow = this.isBelowFrequencyCap(impressions, s);
           if (!isBelow) {
             caps.push(s);
+            this._addSpocsFill(s.id, "frequency_cap", 0, fullRecalc ? 1 : 0);
```

Comment:
I wonder if there's a way to do all the "add calls" inside the transform function? This way it would all be in one place. It's also spoc specific so you wouldn't need to worry about "if it spoc or rec" as many of these functions for filter and score are generic. But I dunno how we would engineer this in that way.

Otherwise, we just have a bunch of ifs and spoc fill stuff happening all over.

Reply:
Agreed. Let me see if I can find a way to do so.

Reply:
If you don't see an easy way, don't sweat it too much. I think the solution you used still works, so if anything other than what you did is a lot of work, we can probably keep your existing solution.